### PR TITLE
Rename preset IDs for consistency

### PR DESCRIPTION
**Pull request by mattdangerw**
And start to add rules we would like to follow to our style guide.

### CODE REVIEW

**Comment by mattdangerw** [comment]
Full list of preset names after the rename...

```python
bert_tiny_en_uncased
bert_tiny_en_uncased_sst2
bert_small_en_uncased
bert_medium_en_uncased
bert_base_en_uncased
bert_base_en
bert_base_zh
bert_base_multi
bert_large_en_uncased
bert_large_en
distil_bert_base_en_uncased
distil_bert_base_en
distil_bert_base_multi
roberta_base_en
roberta_large_en
xlm_roberta_base_multi
xlm_roberta_large_multi
```

File: STYLE_GUIDE.md [hunk 1]
```diff
@@ -18,6 +18,29 @@ When a specific abbreviation is very common and is pronounceable (acronym),
...
+
+- The model and preset names should be recognizable to users familiar with the
+  original release. E.g. the model that goes with the "DeBERTaV3" paper should
+  be called `DebertaV3`. A release of a [toxic-bert](https://huggingface.co/unitary/toxic-bert)
+  checkpoint for `keras_nlp.models.Bert`, should include the string
+  `"toxic_bert"`.
+- All preset names should include the language of the pretraining data. If three
+  or more language are supported, the preset name should include `"multi"` (not
+  the single letter "m").
+- If a preset lowercases input for cased based language, the preset name should
```
Comment (jbischof) [inline STYLE_GUIDE.md:35]: cased-based

File: STYLE_GUIDE.md [hunk 2]
```diff
@@ -18,6 +18,29 @@ When a specific abbreviation is very common and is pronounceable (acronym),
...
+  be marked with `"uncased"`.
+- Don't abbreviate size names. E.g. "xsmall" or "XL" in an original checkpoint
+  releases should map to `"extra_small"` or `"extra_large"` in a preset names.
+- No configuration in names. E.g. use "bert_base" instead of
+  "bert_L-12_H-768_A-12".
+
+When in doubt, readability and usability of names should win out!
```
Comment (jbischof) [inline STYLE_GUIDE.md:42]: "When in doubt, readability should win out!" is catchier. Just saying.

**Review by jbischof — APPROVED** [review]
Great! May this be the last preset naming PR!

**Pull request merged**